examples/example-protocol/src/main.rs

```diff
@@ -48,17 +48,20 @@ fp_import! {
     fn import_void_function_empty_result() -> Result<(), u32>;
 
     // Passing primitives:
-    fn import_primitive_bool(arg: bool) -> bool;
-    fn import_primitive_f32(arg: f32) -> f32;
-    fn import_primitive_f64(arg: f64) -> f64;
-    fn import_primitive_i8(arg: i8) -> i8;
-    fn import_primitive_i16(arg: i16) -> i16;
-    fn import_primitive_i32(arg: i32) -> i32;
-    fn import_primitive_i64(arg: i64) -> i64;
-    fn import_primitive_u8(arg: u8) -> u8;
-    fn import_primitive_u16(arg: u16) -> u16;
-    fn import_primitive_u32(arg: u32) -> u32;
-    fn import_primitive_u64(arg: u64) -> u64;
+    fn import_primitive_bool_negate(arg: bool) -> bool;
+    // FIXME: Imported functions get 0.0 instead of the float values when called from a plugin.
+    // This is a bug in wasmer 2.3, but can be worked around by using a one-sized array instead.
+    // See https://github.com/fiberplane/fp-bindgen/issues/180
+    fn import_primitive_f32_add_one(arg: [f32; 1]) -> f32;
+    fn import_primitive_f64_add_one(arg: [f64; 1]) -> f64;
```

Comment:
I think it may make sense to keep the non-array versions too, but simply omit those from the Wasmer test with a comment similar to this one explaining why. Then when we add a Wasmer 3 runtime, it can have reinstate the tests for those.

Reply:
Yea, that's what I was thinking too.

Reply:
But with the "pass floats as integer" angle, you can pass floats around directly (at least as far as the functions *you* will be working with are concerned) without the need for the "one-sized array" hack, which I think could be better.